client/my-sites/customer-home/controller.jsx

```diff
@@ -19,7 +23,23 @@ export default async function ( context, next ) {
 	next();
 }
 
-export function maybeRedirect( context, next ) {
+const refetchLaunchpadScreenOption = async ( siteId, context ) => {
+	return new Promise( ( resolve ) => {
+		const unsubscribe = context.store.subscribe( () => {
+			if ( isRequestingSite( context.store.getState(), siteId ) ) {
+				return;
+			}
+
+			unsubscribe();
+			resolve();
+		} );
+
+		// Trigger a `store.subscribe()` callback
+		context.store.dispatch( requestSite( siteId ) );
+	} );
+};
```

Comment:
The way how `isRequestingSite` is checked here is a bit fragile. What if there is a Redux state update between dispatch of `requestSite` and update of the `isRequestingSite` state? Then `isRequestingSite` is `false` and the promise resolves even before the request started!

The code relies on the fact that the very first thing that the `requestSite` thunk does is that it dispatches the `SITE_REQUEST` action, which modifies the `isRequestingSite` state. That's true today, but can change.

It's more reliable and much simpler to just await the `requestSite` dispatch:
```js
const refetchLaunchpadScreenOption = async ( siteId, context ) => {
  await context.store.dispatch( requestSite( siteId ) );
}
```

Reply:
Very helpful! Thanks for the guidance. I'll update the PR accordingly.

Reply:
[5a692ed](https://github.com/Automattic/wp-calypso/pull/67967/commits/5a692ed84df76ceb61a28624a7d0ccabadbc44a1)